src/transformers/processing_utils.py

```diff
@@ -494,11 +495,22 @@ def save_pretrained(self, save_directory, push_to_hub: bool = False, **kwargs):
                     del attribute.init_kwargs["auto_map"]
 
         # If we save using the predefined names, we can load using `from_pretrained`
+        # plus we save chat_template in itw own file
         output_processor_file = os.path.join(save_directory, PROCESSOR_NAME)
+        output_chat_template_file = os.path.join(save_directory, CHAT_TEMPLATE_NAME)
+
+        processor_dict = self.to_dict()
+        if processor_dict.get("chat_template", None) is not None:
+            chat_template_dictionary = {"chat_template": processor_dict["chat_template"]}
```

Comment:
I would have expected here that we pop `"chat_template"` so that when we save `processor_dict` out, the chat template won't be contained within the json file

Reply:
oh yeah, we should pop. Bettter than the way I did now